src/lib/error/ErrorReport.ts

```diff
@@ -114,11 +115,7 @@ export class ErrorReport extends Error {
     })
   }
 
-  private static getRequestErrorMetadata(err: AxiosError): RequestErrorDetails | null {
-    if (!err.config) {
-      return null
-    }
-
+  private static parseAxiosError(err: AxiosError): RequestErrorDetails {
     const { url, method, headers: requestHeaders, params, data: requestData, timeout: requestTimeout } = err.config
     const { status, statusText, headers: responseHeaders, data: responseData } = err.response || {}
```

Comment:
Couldn't it generate an error if `err.response` is undefined? Since the check to call this function will be true if only `err.config` is defined.

Reply:
Axios requests leave `err.response` undefined when there was a client error on the request (ECONNABORTED, ETIMEOUT, etc), so `err.response == undefined` is a valid request error scenario (guaranteed, I think, to be a client error)